src/crimson/os/seastore/object_data_handler.cc

```diff
@@ -50,8 +50,8 @@ struct extent_to_write_t {
   extent_to_write_t(const extent_to_write_t &) = default;
   extent_to_write_t(extent_to_write_t &&) = default;
 
-  extent_to_write_t(laddr_t addr, bufferlist to_write)
-    : addr(addr), len(to_write.length()), to_write(to_write) {}
+  extent_to_write_t(laddr_t addr, bufferlist to_write, extent_len_t block_size)
+    : addr(addr), len(p2roundup(to_write.length(), block_size)), to_write(to_write) {}
```

Comment:
to_write is required to be block aligned, let's assert that to_write.length() is aligned here and fix the callers.